### PR TITLE
Update CSES Road Construction.mdx

### PR DESCRIPTION
**Pull request by TruaShamu**
Continued from the [previous pull request](https://github.com/cpinitiative/usaco-guide/pull/1277):
Added implementation heading and time complexity.

### CODE REVIEW

File: solutions/CSES Road Construction.mdx [hunk 1]
```diff
@@ -54,9 +56,10 @@ int main () {
 
 <JavaSection>
 
-```java
-
+**Time Complexity:** $\mathcal{O}(N\log N)$
```
Comment (Farrius) [inline solutions/CSES Road Construction.mdx:59]: Put this before "LanguageSection" I think it can be the reason why the build is not working.

**Review by Farrius — CHANGES_REQUESTED** [review]

File: solutions/CSES Road Construction.mdx (hunk 1 above)
Comment (maggie-j-liu) [reply solutions/CSES Road Construction.mdx]: I think this is fine, it seems that the prettier check is failing.

Comment (Farrius) [reply solutions/CSES Road Construction.mdx]: Do you know how to solve that?

Comment (thecodingwizard) [reply solutions/CSES Road Construction.mdx]: ehh just ignore it, I think I'm going to get rid of the prettier check tbh. It's causing a lot more trouble than it's actually helping imo

**Pull request merged**